enterprise/internal/batches/background/batch_spec_workspace_execution_worker.go

```diff
@@ -122,8 +147,11 @@ func (s *batchSpecWorkspaceExecutionWorkerStore) MarkComplete(ctx context.Contex
 
 	job, changesetSpecIDs, err := loadAndExtractChangesetSpecIDs(ctx, tx, int64(id))
 	if err != nil {
-		// If we couldn't extract the changeset IDs, we mark the job as failed
-		return s.Store.MarkFailed(ctx, id, fmt.Sprintf("failed to extract changeset IDs ID: %s", err), options)
+		return s.MarkFailed(ctx, id, fmt.Sprintf("failed to extract changeset IDs ID: %s", err), options)
+	}
+
+	if err := resetAndDeleteAccessToken(ctx, tx, int64(id)); err != nil && err != store.ErrNoResults {
+		return s.MarkFailed(ctx, id, fmt.Sprintf("failed to delete internal access token: %s", err), options)
```

Comment:
I think we should handle this here, too.

```suggestion
	if err := resetAndDeleteAccessToken(ctx, tx, int64(id)); err != nil && err != store.ErrNoResults && err != database.ErrAccessTokenNotFound {
		return s.MarkFailed(ctx, id, fmt.Sprintf("failed to delete internal access token: %s", err), options)
```

Reply:
Also, should we copy these error conditions to markFailed and markErrored, too?

Reply:
> I think we should handle this here, too.

Yeah, let's see if we can use `errcode.IsNotFound` here.